pkgs/development/interpreters/clojurescript/lumo/package.json

```diff
@@ -0,0 +1,110 @@
+{
+  "name": "lumo-cljs",
```

Comment:
Would it be possible to add `lumo-cljs` to `pkgs/development/node-packages/node-packages-v8.json`
Then we can share its expression with other node packages. 17k of dependencies is not acceptable for a single package. Git and nix do not like large files like this.

Reply:
Another thing, if two nix epression want to fetch the same url with the same sha sum, isn't that resource shared in nix-store, even tough the reference is seperate?

Reply:
It's also not 17k, it's 1038 fetchurl's to npm registry. It's a whole lot I admit.

Reply:
It is not just about sharing the source but also evaluation time and channel size (each user would then need to download 1mb extra just for a single package).

Reply:
If you the package would be included in `nodePackages` can its dependencies not also used from there?

```console
$ nix repl '<nixpkgs>'
nixpkgs> :b nodePackages.eslint
> "${nodePackages.eslint}"
/nix/store/7sk5kg95rdy25rjdxvknxl3rmkgjcxxg-node-eslint-5.8.0
> ls -la /nix/store/7sk5kg95rdy25rjdxvknxl3rmkgjcxxg-node-eslint-5.8.0/lib/node_modules/eslint/node_modules/
```

We could even delete unusable binaries from the package itself by adding an override.
Many of the dependencies listed here are already include in node-packages.

Reply:
if the package is in node dependencies, I will have to test if the build tool (yarn/npm) will no just try to make lockfile, or fetch dep graph fro the mirror. I had this problem with npm2nix, that I couldn't stop it from making http requests. The thing that I worry about too, is that I would only be adding lumo-cljs to nix's node-packages for its build dependencies but not runtime, and the readme in `pkgs/development/node-packages` said it only wants top-level packages. Lumo isn't a traditional package, npm is part of the process of building it, it serves more as a way to install the binary globaly than as node module. You are right that I can diminsih the size quite a bit, it's now total for 6 files 633.6kb and I made a comment about this size in my PR message. I could try to make this smaller. And I can try adding lumo-cljs as non top-level npm module.

Reply:
"Libraries should only be added to the package set if there is a non-NPM package that requires it." so in the Lumo's case it should be fine :)

Reply:
We do the same already for meguca.

Reply:
Can you look over my PR now, this looks quite strange, the version name actually takes relative path, and that's appended to the nodePackages attribute name. But it works fine like this (hopefully GrahamcOfBorg aggrees).